### PR TITLE
[MM-16513] Migrate Token.Delete to Sync by default

### LINKED ISSUE #11347: Migrate "Token.Delete" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-16513)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `Delete` in the `Token` store.

The expected way to implement it is, go to the `Token` store implementation in the `store/sqlstore/` directory, modify the method `Delete` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### PR DESCRIPTION
**Pull request by goku321**
#### Summary

Migrate Token.Delete to Sync by default
#### Ticket Link

Fixes #11347

### CODE REVIEW

File: app/user.go [hunk 1]
```diff
@@ -1362,8 +1362,8 @@ func (a *App) GetPasswordRecoveryToken(token string) (*model.Token, *model.AppEr
 }
 
 func (a *App) DeleteToken(token *model.Token) *model.AppError {
-	if result := <-a.Srv.Store.Token().Delete(token.Token); result.Err != nil {
```
Comment (jespino) [inline app/user.go]: Here you can just `return a.Srv.Store.Token().Delete(token.Token)`

Comment (goku321) [reply app/user.go]: Thanks for the suggestion. I have made the changes.

**Comment by DSchalla** [comment]
@goku321 Sorry for the late review! Can you merge the current master to fix the conflicts before we merge? :-)

**Comment by goku321** [comment]
Sure @DSchalla will do :)

**Comment by jespino** [comment]
Merged, thanks! 🎉

**Pull request merged**